### ISSUE TITLE
java.lang.IllegalArgumentException: onFileSavedAs callback is NULL

### ISSUE THREAD

**Issue by Nico04** [LaemonT/simple_file_saver#1]
I've got a weird bug, when using `Share.shareXFiles` of `share_plus` package on Android 14, it throws and closes the app completly after the native sharing bottom sheet closes, with an error related to this package (the method that shares do NOT use the package)

```
D/AndroidRuntime(12746): Shutting down VM
E/AndroidRuntime(12746): FATAL EXCEPTION: main
E/AndroidRuntime(12746): Process: ***, PID: 12746
E/AndroidRuntime(12746): java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=22643, result=0, data=null} to activity {***.MainActivity}: java.lang.IllegalArgumentException: onFileSavedAs callback is NULL
E/AndroidRuntime(12746): 	at android.app.ActivityThread.deliverResults(ActivityThread.java:5527)
E/AndroidRuntime(12746): 	at android.app.ActivityThread.handleSendResult(ActivityThread.java:5566)
E/AndroidRuntime(12746): 	at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:67)
E/AndroidRuntime(12746): 	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
E/AndroidRuntime(12746): 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:139)
E/AndroidRuntime(12746): 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:96)
E/AndroidRuntime(12746): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2443)
E/AndroidRuntime(12746): 	at android.os.Handler.dispatchMessage(Handler.java:106)
E/AndroidRuntime(12746): 	at android.os.Looper.loopOnce(Looper.java:205)
E/AndroidRuntime(12746): 	at android.os.Looper.loop(Looper.java:294)
E/AndroidRuntime(12746): 	at android.app.ActivityThread.main(ActivityThread.java:8177)
E/AndroidRuntime(12746): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime(12746): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
E/AndroidRuntime(12746): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971)
E/AndroidRuntime(12746): Caused by: java.lang.IllegalArgumentException: onFileSavedAs callback is NULL
E/AndroidRuntime(12746): 	at com.laemont.simple_file_saver.FileSaveManager.onActivityResult(FileSaveManager.kt:105)
E/AndroidRuntime(12746): 	at io.flutter.embedding.engine.FlutterEngineConnectionRegistry$FlutterEngineActivityPluginBinding.onActivityResult(FlutterEngineConnectionRegistry.java:774)
E/AndroidRuntime(12746): 	at io.flutter.embedding.engine.FlutterEngineConnectionRegistry.onActivityResult(FlutterEngineConnectionRegistry.java:422)
E/AndroidRuntime(12746): 	at io.flutter.embedding.android.FlutterActivityAndFragmentDelegate.onActivityResult(FlutterActivityAndFragmentDelegate.java:857)
E/AndroidRuntime(12746): 	at io.flutter.embedding.android.FlutterFragment.onActivityResult(FlutterFragment.java:1251)
E/AndroidRuntime(12746): 	at io.flutter.embedding.android.FlutterFragmentActivity.onActivityResult(FlutterFragmentActivity.java:653)
E/AndroidRuntime(12746): 	at android.app.Activity.dispatchActivityResult(Activity.java:8943)
E/AndroidRuntime(12746): 	at android.app.ActivityThread.deliverResults(ActivityThread.java:5520)
E/AndroidRuntime(12746): 	... 13 more
I/Process (12746): Sending signal. PID: 12746 SIG: 9
```

Look similar to this issue https://github.com/fluttercommunity/plus_plugins/issues/1374

EDIT : it happens when using other native packages like flutter_downloader

**Comment by Nico04** [comment]
It's probably comming from [this line](https://github.com/LaemonT/simple_file_saver/blob/c5d17a7c58b89f7a3e9f62328d2a5f6681640f8b/simple_file_saver_android/android/src/main/kotlin/com/laemont/simple_file_saver/FileSaveManager.kt#L105) ?

**Comment by LaemonT** [comment]
> It's probably comming from [this line](https://github.com/LaemonT/simple_file_saver/blob/c5d17a7c58b89f7a3e9f62328d2a5f6681640f8b/simple_file_saver_android/android/src/main/kotlin/com/laemont/simple_file_saver/FileSaveManager.kt#L105) ?

Thanks for your information, I think all the overridden "onActivityResult" method is triggered whenever a native package opens an activity and receives a result. Since they are registered to the same FlutterActivity.

It is now fixed in 1e4be01.
